### PR TITLE
Update error message when dataset isn't configured with time_column in append refresh

### PR DESCRIPTION
**Pull request by Sevenannn**
## 🗣 Description

A minor update to error message when dataset doesn't support append refresh mode without time column. Before the error message is misleading as it indicates the connector engine doesn't support refresh mode, whereas it's only because `time_column` is not configured.

Before:
image

After:
image

### CODE REVIEW

File: crates/runtime/src/datafusion.rs [hunk 1]
```diff
@@ -197,8 +197,10 @@ pub enum Error {
     #[snafu(display("{source}"))]
     InvalidTimeColumnTimeFormat { source: refresh::Error },
 
-    #[snafu(display("Acceleration mode {mode} not supported for dataset from source {from}"))]
-    UnsupportedAccelerationMode { mode: String, from: String },
+    #[snafu(display(
+         "Acceleration mode `append` requires `time_column` parameter for source {from}.\nConfigure `time_column` parameter with a valid value.\nFor details, visit: https://docs.spiceai.org/reference/spicepod/datasets#time_column"
```
Comment (sgrebnov) [inline crates/runtime/src/datafusion.rs]:
```suggestion
         "Acceleration mode `append` requires `time_column` parameter for source {from}.\nConfigure `time_column` parameter and try again.\nFor details, visit: https://docs.spiceai.org/reference/spicepod/datasets#time_column"
```

**Review by sgrebnov — DISMISSED** [review]

File: crates/runtime/src/datafusion.rs [hunk 2]
```diff
@@ -197,8 +197,10 @@ pub enum Error {
...
+    ))]
+    UnsupportedAppendAcceleration { from: String },
```
Comment (sgrebnov) [inline crates/runtime/src/datafusion.rs]:
```suggestion
    AppendRequiresTimeColumn { from: String },
```

**Comment by Jeadie** [comment]
Does this also fix https://github.com/spiceai/spiceai/issues/3702

**Comment by Sevenannn** [comment]
https://github.com/spiceai/spiceai/issues/3702 is a different error fixing issue, spiceai connector supports append refresh without time column. We should provide users a warning (or something equivalent) when they are doing refresh without time column, telling user the `time_column` param won't be used in this case.

**Review by sgrebnov — APPROVED** [review]

**Pull request merged**